doc/bu-xthin-protocol.md

```diff
@@ -0,0 +1,382 @@
+
+- The coinbase transaction will always be included in the thinblock.
+
+- During startup when the memory pool has few transactions in it, or when a block is very small and has only 1 or 2 transactions a 
+thinblock may end up being larger than the regular block. In that case a regular block will be returned to the requestor instead of a 
+thinblock. This typically happens when a new block is mined just seconds after the previous one.
+
+- Bloom Size Decay algorithm: A useful phenomena occurs as the memory pools grow and get closer in sync; the bloom filter can be allowed 
+to become less sparse. That means more false positives but because the memory pool has been �warmed up� there is now a very low likelihood 
+of missing a transaction. This bears out in practice and a simple linear decay algorithm was developed which alters both the number of 
+elements and the false positive rate. However, not knowing how far out of sync our pools are in practice means we can not calculate the 
+with certainty the probability of a false positive and a memory pool miss which will result in a re-requested transaction, so we need to 
+be careful in not cutting too fine a line. Using this approach significantly reduces the needed size of the bloom filter by 50%.
+
+- Bloom Filter Targeting (Beginning in version 12.1): Bloom Filter targeting which is effective in reducing the size of the bloom filters 
+during times when the memory pool is over run with transactions.  This targeting process is memory pool size independent and yields reliably
+smaller bloom filters. It works essentially by narrowing down the transactions to include in the bloom filter to just those that are most 
+likely to be mined in the next block.
+
+- A 64 bit transaction hash is used instead of the full 256 bit hash to further reduce thinblock size while still preventing hash collisions 
+in the memory pool.  This is done using the GET_XTHIN p2p netmessage.  If the block can not be constructed with the transactions in the mempool
+then a GET_XBLOCKTX netmessage will be requested and will contain a list of the missing transaction hashes to be retreieved.  The returning
+XBLOCKTX message will contain a list of the transactions requested.
+
+- In the event of a 64 bit hash collision the receiving peer will re-request a normal THINBLOCK with the full 256 bit transaction hashes.
+This is done using a GETDATA with an inventory message containing the MSG_THINBLOCK message type.
+
+- In the event that the re-requested THINBLOCK containing the full transaction hashes is not received or is missing transactions then a
+normal block is requested.
+
+
+3. Detailed Protocol Specification
+----------------------------------
+----------------------------------
+
+
+XTHIN Service Bit
+-----------------
+
+defined in protocol.h.  Add the following service:
+
+    /** nServices flags */
+    enum {
+
+        NODE_XTHIN = (1 << 4),
+
+    ... }
```

Comment:
No need to mention where the enum is defined. It should be enough just to say:

"a node signals that it supports serving xthin blocks using sevice bit 

`NODE_XTHIN = (1 << 4)`

"